### ISSUE TITLE
Differentiability Issue -> why recursive loop loss isn't effective

### ISSUE THREAD

**Issue by ramithuh** [ramithuh/differentiable-trees#19]
It seems that I had sliced based on the most probable paths that connect to a parent node.  (code line below)
This does not help during training.

https://github.com/ramithuh/differentiable-trees/blob/8570b49d030b68b7a0b1b57ebc42c24d506348a7/modules/tree_func.py#L199

Need to look at : 
- [ ] - [1] - https://papers.nips.cc/paper/2021/file/89ae0fe22c47d374bc9350ef99e01685-paper.pdf
- [ ] - [2] - https://arxiv.org/pdf/1912.02175.pdf
- [ ] [code] https://github.com/martius-lab/blackbox-backprop/blob/0152a6df8c819580d800b052ef17f39bdb82eb98/blackbox_backprop/shortest_path.py#L69

There is a question as to whether making traversal differentiable makes sense considering the scaling aspect as mentioned in [1].
excerpts : https://arc.net/e/E7811694-AD1C-40C0-8BA9-4714BDFB5877

**Comment by ramithuh** [comment]
- [ ]  [Learning with Differentiable Perturbed Optimizers](https://arxiv.org/abs/2002.08676)

**Comment by ramithuh** [comment]
- [ ] [Neural Distance Embeddings for Biological Sequences](https://arxiv.org/abs/2109.09740)
